From ba6b3dfbe59a874911343663702def2e11959b43 Mon Sep 17 00:00:00 2001
From: Phil Smith <phil.h.smith@gmail.com>
Date: Sat, 20 Mar 2010 16:38:23 -0700
Subject: [PATCH] Generated models include table_name_prefix if they have enclosing namespaces. [#4230 state:resolved]

This makes them use the tables their migrations create.  For example,
"rails g model gallery/image" will make

class Gallery::Image < ActiveRecord::Base
  self.table_name_prefix = "gallery_"
end
---
 .../active_record/model/templates/model.rb         |    3 +++
 railties/test/generators/model_generator_test.rb   |   11 +++++++++++
 2 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/generators/active_record/model/templates/model.rb b/activerecord/lib/generators/active_record/model/templates/model.rb
index 21ae29e..ae9088e 100644
--- a/activerecord/lib/generators/active_record/model/templates/model.rb
+++ b/activerecord/lib/generators/active_record/model/templates/model.rb
@@ -1,4 +1,7 @@
 class <%= class_name %> < <%= parent_class_name.classify %>
+<% unless class_path.empty? -%>
+  self.table_name_prefix = "<%= class_path.join('_') %>_"
+<% end -%>
 <% attributes.select {|attr| attr.reference? }.each do |attribute| -%>
   belongs_to :<%= attribute.name %>
 <% end -%>
diff --git a/railties/test/generators/model_generator_test.rb b/railties/test/generators/model_generator_test.rb
index 79ce9a2..1f1ecca 100644
--- a/railties/test/generators/model_generator_test.rb
+++ b/railties/test/generators/model_generator_test.rb
@@ -27,6 +27,17 @@ class ModelGeneratorTest < Rails::Generators::TestCase
     assert_file "app/models/account.rb", /class Account < Admin::Account/
   end
 
+  def test_model_with_namespace
+    run_generator ["Gallery::Image"]
+    assert_file "app/models/gallery/image.rb", /class Gallery::Image/
+    assert_file "app/models/gallery/image.rb", /self.table_name_prefix = "gallery_"/
+  end
+
+  def test_model_with_nested_namespace
+    run_generator ["Admin::Gallery::Image"]
+    assert_file "app/models/admin/gallery/image.rb", /self.table_name_prefix = "admin_gallery_"/
+  end
+
   def test_migration
     run_generator
     assert_migration "db/migrate/create_accounts.rb", /class CreateAccounts < ActiveRecord::Migration/
-- 
1.6.6

